src/components/Netcat/Netcat.tsx

```diff
@@ -30,19 +22,60 @@ interface FormValuesType {
     const [selectedScanOption, setSelectedNetcatOption] = useState("");
-    const [pid, setPid] = useState("");
-    const [loading, setLoading] = useState(false);
-    const [allowSave, setAllowSave] = useState(false);
-    const [hasSaved, setHasSaved] = useState(false);
-    const [checkedVerboseMode, setCheckedVerboseMode] = useState(false);
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [allowSave, setAllowSave] = useState(false); // State variable to allow saving the output to a file
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved
+    const [checkedVerboseMode, setCheckedVerboseMode] = useState(false); // State variable to indicate whether the verbose mode is enabled.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state of the modal.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the modal is opened.
+    // Component Constants.
+    const title = "Netcat"; // Title of the component.
+    const description =
+        "A simple Unix utility which reads and writes data across network connections using TCP or UDP protocol."; // Description of the component.
+    const steps =
+        +"Step 1: If you want to listen for connections for chat or reverse shell choose the listen option and provide a port number.\n" +
+        "Step 2: If you want to scan for ports, provide an IP address and a port range\n" +
+        "Step 3: If you want to send a file, provide the destination IP address, port number, and File name\n" +
+        "Step 4: If you want to receive a file, provide a port number and the file name.\n" +
+        "Step 5: If you want to port scan a domain, provide Domain name and a port number. \n\n" +
+        "Note:   You should only use website port scan to a domain that you own.\n" +
+        "Note 2: Using the sending/receiving file option might seem like it is not working, but it is working.\n" +
+        "Note 3: You will need two devices for file transfer to work — one for sending and the other need to be set up on the specified port to capture the incoming file.\n" +
+        "        Ensure both machines are properly configured and connected to the same network or through an accessible route (such as a VPN, or through valid public IP addresses) to complete the file transfer.\n";
+    const sourceLink = "https://www.kali.org/tools/netcat/"; // Link to the source code
+    const tutorial = ""; // Link to the official documentation/tutorial.
+    const dependencies = ["netcat-traditional"]; // Contains the dependencies required by the component
```

Comment:
Sorry Oliver this is my bad. It should reflect the command, therefore it should be nc. :-(
```suggestion
    const dependencies = ["nc"]; // Contains the dependencies required by the component
```